backend/pkg/transport/network/tcp/tcp_test.go

```diff
@@ -0,0 +1,86 @@
+package tcp_test
+
+import (
+	"encoding/json"
+	"github.com/HyperloopUPV-H8/h9-backend/pkg/transport/network/tcp"
+	"github.com/rs/zerolog"
+	"os"
+	"testing"
+	"time"
+)
+
+type Socket struct {
+	SrcIP   string
+	SrcPort uint16
+	DstIP   string
+	DstPort uint16
+}
+
+// Payload defines a piece of information comming from the network with its metadata
+type Payload struct {
+	Timestamp time.Time
+	Socket    Socket
+	Data      []byte
+}
+
+func TestTCP(t *testing.T) {
+	writer := os.Stdout
+	logger := zerolog.New(writer)
+
+	// Server setup
+	serverAddr := "127.0.0.1:3000"
+	server := tcp.NewServer(serverAddr, tcp.ServerConfig{}, logger)
+	go func() {
+		if err := server.Listen(); err != nil {
+			t.Fatalf("Server failed to listen: %v", err)
+		}
+	}()
+	time.Sleep(1 * time.Second)
+
+	// Client setup
+	client := tcp.NewClient(serverAddr, tcp.ClientConfig{}, logger)
+	conn, err := client.Dial()
+	if err != nil {
+		t.Fatalf("Client failed to dial: %v", err)
+	}
+	defer conn.Close()
+
+	// Create and send Payload
+	originalPayload := Payload{
+		Timestamp: time.Now(),
+		Socket: Socket{
+			SrcIP:   "127.0.0.1",
+			SrcPort: 3000,
+			DstIP:   "127.0.0.1",
+			DstPort: 3000,
+		},
+		Data: []byte("hello"),
+	}
+	payloadBytes, err := json.Marshal(originalPayload)
+	if err != nil {
+		t.Fatalf("Failed to serialize Payload: %v", err)
+	}
```

Comment:
if the objective is just to test if the client and server connection correctly read and write, why not just use a plain `[]byte` array initialized to some string? It takes complexity out of the test and removes a possible cause of errors